platform/javascript/display_server_javascript.cpp

```diff
@@ -407,23 +407,23 @@ void DisplayServerJavaScript::cursor_set_custom_image(const RES &p_cursor, Curso
 
 // Mouse mode
 void DisplayServerJavaScript::mouse_set_mode(MouseMode p_mode) {
-	ERR_FAIL_COND_MSG(p_mode == MOUSE_MODE_CONFINED, "MOUSE_MODE_CONFINED is not supported for the HTML5 platform.");
+	ERR_FAIL_COND_MSG(p_mode & MOUSE_MODE_CONFINED, "MOUSE_MODE_CONFINED is not supported for the HTML5 platform.");
 	if (p_mode == mouse_get_mode())
 		return;
 
 	if (p_mode == MOUSE_MODE_VISIBLE) {
 		godot_js_display_cursor_set_visible(1);
 		emscripten_exit_pointerlock();
 
-	} else if (p_mode == MOUSE_MODE_HIDDEN) {
-		godot_js_display_cursor_set_visible(0);
-		emscripten_exit_pointerlock();
-
-	} else if (p_mode == MOUSE_MODE_CAPTURED) {
+	} else if (p_mode & MOUSE_MODE_CAPTURED_VISIBLE) {
```

Comment:
Why captured visible? Pointer should be locked when in `CAPTURED` mode even when if invisible. Actually, I think that by default, it turns it invisible.

Reply:
`MOUSE_MODE_CAPTURED` is a union (`3`) of the captured state (`2`) and the hidden state (`1`). `MOUSE_MODE_CAPTURED_VISIBLE` is just the captured state without the hidden state (so it's visible). Captured should usually imply hidden, which is why I made the exposed `MOUSE_MODE_CAPTURED` have the hidden state.

Reply:
this is pretty confusing, but I guess the implementation is fine then.

Reply:
Maybe it would be less confusing if it was named `MOUSE_MODE_CAPTURED_ONLY`?